sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusAsyncConsumer.java

```diff
@@ -26,17 +30,28 @@
     private final String linkName;
     private final ServiceBusReceiveLinkProcessor linkProcessor;
     private final MessageSerializer messageSerializer;
-    private final Flux<ServiceBusReceivedMessage> processor;
+    private final Flux<ServiceBusReceivedMessage> messageSource;
 
     ServiceBusAsyncConsumer(String linkName, ServiceBusReceiveLinkProcessor linkProcessor,
-        MessageSerializer messageSerializer, int prefetch) {
+        MessageSerializer messageSerializer, int prefetch, boolean autoLockRenewal,
+        Duration maxAutoLockRenewDuration, LockContainer<LockRenewalOperation> messageLockContainer,
```

Comment:
Can we keep this to ReceiveOptions? it seems like we're bloating this constructor

Reply:
I'm also adding another parameter from ReceiveOptions.

Reply:
Okay but I am not adding `messageLockContainer` since that is not related to ReceiveOptions.